### ISSUE TITLE
Source Code?

### ISSUE THREAD

**Issue by n9yty** [gidodidi/BMBF#3]
I'm sure the omission was itnentional, so I am guessing this is not open source, but I thought it wouldn't hurt to ask.

**Comment by ryan77627** [comment]
My guess would have to be it's a fork of BeatOn since the GUI is almost the same and all of the components are still available, just not the combination of them into BeatOn. Or maybe they had a fork from earlier.

**Comment by curvatura99** [comment]
If this is a fork of BeatOn, which is most likely the case, then the dev is violating the MIT license since he/she did not distribute notice with the software.

**Comment by pnoeric** [comment]
Indeed, if it's a fork of BeatOn, why would the developers not share what they're doing? I love the functionality but slightly concerned that we can't see the source... they could be hiding something (anything) in the installers and we'd never know.

**Comment by curvatura99** [comment]
>  they could be hiding something (anything) in the installers and we'd never know.

Well apk files are pretty easy to de-compile, but we shouldn't need to do that. Now if the original license was GPL then this would have been a non-issue.

**Comment by pnoeric** [comment]
Aha, ok. I'm glad to know APK files are easy to look into. That makes me feel better.

But in general, I wish it was open source. I am unclear why the developer would not make it open source (especially if it's based on another open-source project)... the whole point is to let others contribute. I'm nervous when I see a project like this go from open source to closed. It's not like they're selling it or protecting some trade secret. Unless it's the method in which they modify Beat Saber, and they're worried that if the BS developers see this method they'll patch the hole. If that's the case, I'd like for them to say so. Or make the rest of the project open-source and just keep that part private.

**Comment by shaneharris** [comment]
I think its pretty clear why the source hasnt been released, the developer clearly doesnt use this repo for development, this is a burner github account. The developer clearly doesnt want to be traced back to the app which makes conventional version control pretty impossible. I woukd say if the code is released then it will be a zip of the files, not a full version history. Having the source like that might not give you any more confidence than no source at all because you have to trust that is the source of the current binary release - something you cant verify without reverse engineerimg the apps anyway. I guess that mentality applies to any binary release though, you have to trust that the compiled minified version is indeed based on the same source code

**Comment by pnoeric** [comment]
(Let's put aside the trust issue and assume that there is nothing nefarious going on here.)

For a project like this, going open source (and allowing the community to contribute) is a HUGE benefit to the developer and all the users.... it's foolish to not go open source; so foolish that again, it makes me wonder exactly why, since there are very few reasons not to do so.

IMHO if the developer doesn't want to be traced back to the app, then run the project with a burner email address attached to the GitHub account. Exposing the source code (especially if it's based on a previous project that is/was open source?) is the right way to proceed. Or-- just put the source code in a zip file and dump it somewhere and let other people pick it up and work on it and manage whatever risk is involved on their own GitHub account. Then the developer can just write the code and set it down and walk away.

Anyhow. It's all good. I like the app and I'm grateful to whomever created it. I'm just surprised and disappointed that they didn't see any path to release the source, and harness the energy and knowledge of the community so we can all make it better. Open source isn't a passing fad; it's how projects like this are done these days. (See: rutorrent, sickrage/sickchill, couch potato, angular, jquery, linux, npm, etc.)

**Comment by shaneharris** [comment]
I think HUGE is a bit of an over statement. I dont know how many open source projects you have run but even my most popular ones struggle to get the contribution they need. Contribution isnt a free flowing as you suggest and often people only open PRs for small modifications they want or for simple housekeeping.

The idea is nice, but in reality there isnt a HUGE incentive  for the developer to release the source, its just the right thing to do in most cases. In this case, maybe not. 

If you dont trust this dev, how can you trust a zip of source from the developer without reverse engineering the binaries anyway?

**Comment by pnoeric** [comment]
Sure, I hear you. Yet, if they don't offer it as open-source, they'll get 0 contributions, and they miss out on the work of everyone who does want to help but is unable to do so. Worst case, the source code is available, nobody contributes, but still: others can look at and learn from the code. People can fork it and make their own versions. Documentation can be offered/improved (something badly needed and where I want to contribute). Etc.

You said that releasing the code is the "right thing to do in most cases." I agree completely. But then you said "in this case, maybe not." The _only_ maybe-not I've heard is vague concerns about possible legal issues that seem trivial to work around. (In addition to the ideas I mentioned earlier in this thread, here's another, release the code "for educational purposes only.")

In all other cases, it seems like a good idea.

And hey, I'm not trying to argue or "win" and I definitely don't want to get into some stupid internet flame war... life's too short. ;-) I'm just trying to make the point (that hopefully the developer will see, btw!) that I can see a lot of positives for releasing the code and very few negatives. So far I've heard one possible-negative that sounds pretty weak to me. If there are others, let me know, I'm open to understanding more if I'm missing something.

**Comment by curvatura99** [comment]
@shaneharris You seem to know a lot about the mindset of the developer. You original post sounds just like someone defending against a personal attack. So when is the source going to be released?

**Comment by nike4613** [comment]
@curvatura It's entirely possible to defend someone based on presumed thoughts based on behaviour. (See all of us modders on Beat Games.) As much as I wish we knew more about their actions, we can only speculate, but we can make educated guesses and discuss/argue accordingly. Same thing applies here. 

We know *nothing* about the devs, and so are doing our best with the info we have, being that this is a burner account, posting only binaries. With the additional context of why emulamer originally dropped BeatOn, we can guess that the devs are doing this to protect from that worry as completely as possible.

**Comment by nike4613** [comment]
In addition, as modders ourselves (or curators perhaps), we feel the need to communicate our view of the situation and defend them as we feel they are doing this for good reason.

**Comment by pnoeric** [comment]
To be clear - I'm not attacking anyone, so I don't think "defending" is necessary. My point (as I've stated repeatedly) is simply that I can't see a good reason for the dev to _not_ make the source code available, and I can see many good reasons why that would be great for the dev, the project, the community. 

We are all on the same team and all want to do cool things with Beat Saber and our VR rigs. That's the spirit of me asking why the source code isn't available. That's all.

And if @shaneharris is the developer, that's awesome, then he's reading what I've written here and hopefully thinking about it. BTW if you haven't already, check out Shane's other repos... he's doing some _VERY_ cool VR stuff... whether he's the BMBF developer or not, he's a dev to watch, for sure.

**Comment by shaneharris** [comment]
@curvatura99 well given that I was a big contributor to BeatOn and fully understand that situation then yeh I guess I have some insight into the developers mindset. If you are suggesting I am the developer, I don't think its humanly possible for me to be the developer considering how much workload I have with SideQuest. So to answer your question, I have no idea when the source is coming out as I'm not the developer. I am as eager as anyone to see it, given that the desktop installer is clearly a fork of SideQuest I would be the best person to be able to determine if there is anything dodgy in there.
